src/System.Slices/System/ReadOnlySpan.cs

```diff
@@ -338,5 +338,19 @@ public override bool Equals(object obj)
             }
             return false;
         }
+
+        public static bool operator ==(ReadOnlySpan<T> span1, ReadOnlySpan<T> span2) => span1.Equals(span2);
+        public static bool operator !=(ReadOnlySpan<T> span1, ReadOnlySpan<T> span2) => !span1.Equals(span2);
+        public bool Equals(Span<T> other) => other.ReferenceEquals(Object, Offset, Length);
+
+        public static bool operator ==(ReadOnlySpan<T> span, T[] array)
+        {
+            return span.Object == array && 
+                span.Length == array.Length && span.Offset == new UIntPtr((uint)SpanHelpers<T>.OffsetToArrayData);
```

Comment:
Why not use `ReferenceEquals(object, UIntPtr, int)` here? Will keep core equality logic in one place.